pkgs/tools/admin/boulder/default.nix

```diff
@@ -1,50 +1,152 @@
-{ lib
-, fetchFromGitHub
-, buildGoModule
-, testers
-, boulder
+{
+  lib,
+  fetchFromGitHub,
+  buildGoModule,
+  testers,
+  boulder,
 }:
 
 buildGoModule rec {
   pname = "boulder";
-  version = "2022-09-29";
+  version = "2024-06-17a";
 
   src = fetchFromGitHub {
     owner = "letsencrypt";
     repo = "boulder";
     rev = "release-${version}";
     leaveDotGit = true;
     postFetch = ''
-      cd $out
+      pushd $out
       git rev-parse --short=8 HEAD 2>/dev/null >$out/COMMIT
-      find "$out" -name .git -print0 | xargs -0 rm -rf
+      find $out -name .git -print0 | xargs -0 rm -rf
+      popd
     '';
-    hash = "sha256-MyJHTkt4qEHwD1UOkOfDNhNddcyFHPJvDzoT7kJ2qi4=";
+    hash = "sha256-kObCD9diy1ryyeLQNyfWNMJPfvtjAWVp8OVUO0MLV6A=";
   };
 
   vendorHash = null;
 
   subPackages = [ "cmd/boulder" ];
 
-  patches = [ ./no-build-id-test.patch ];
-
   ldflags = [
     "-s"
     "-w"
     "-X github.com/letsencrypt/boulder/core.BuildHost=nixbld@localhost"
   ];
 
   preBuild = ''
-    ldflags+=" -X \"github.com/letsencrypt/boulder/core.BuildID=${src.rev} +$(cat COMMIT)\""
+    ldflags+=" -X \"github.com/letsencrypt/boulder/core.BuildID=${version} +$(cat COMMIT)\""
     ldflags+=" -X \"github.com/letsencrypt/boulder/core.BuildTime=$(date -u -d @0)\""
   '';
 
   preCheck = ''
     # Test all targets.
     unset subPackages
 
+    # Skip tests that fail.
```

Comment:
This is just too much to maintain. Maybe you can skip individual tests by setting something like `-skip '<regexp>'` to `$GOFLAGS` in [the check phase](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/go/module.nix#L292). 

Something like:
```nix
preCheck = ''
  unset subPackages
  export GOFLAGS="''${GOFLAGS}:--skip 'TestRedis.*|TestUpdate'}"
''
```

But that also is going to be a looong list...

Reply:
Maybe try wrapping it with `concatStringsSep "|" ...` so that this regexp can be managed by a Nix list. 

We probably should include something like this in the Go stdenv builder itself in the future.

Reply:
Thanks for the suggestion, @azahi, I'll l👀k into it.

Reply:
After giving it some thought, @azahi, I think maintainability here is more important than skipping tests instead of removing them.

I went ahead and removed commit  https://github.com/NixOS/nixpkgs/commit/cc2ad6e42585c1ad75193ce95ce37f7d0bdd3064 from this PR that introduced skipping tests.